Modules/CPV/include/CPV/PhysicsTask.h

```diff
@@ -52,14 +52,28 @@ class IntensiveTH2F : public TH2F, public o2::mergers::MergeInterface
   }
 
   void setCycleNumber(uint32_t cycleNumber) { mCycleNumber = cycleNumber; }
+  uint32_t getCylceNumber() { return mCycleNumber; }
 
   void merge(MergeInterface* const other) override
   {
+    LOG(debug) << GetName() << " IntensiveTH2F::merge() is being called. this.mCycleNumber = " << mCycleNumber
+               << "; other->mCycleNumber = " << dynamic_cast<IntensiveTH2F*>(other)->mCycleNumber;
     if (mCycleNumber < dynamic_cast<IntensiveTH2F*>(other)->mCycleNumber) {
-      LOG(info) << "IntensiveTH2F::merge() :" << GetName() << "is updating!";
-      *this = *(dynamic_cast<IntensiveTH2F*>(other));
+      LOG(debug) << "IntensiveTH2F::merge() :" << GetName() << "is updating!";
+      //*this = *(dynamic_cast<IntensiveTH2F*>(other));
+      mCycleNumber = dynamic_cast<IntensiveTH2F*>(other)->mCycleNumber;
+      LOG(debug) << "other.fN = " << (*(dynamic_cast<IntensiveTH2F*>(other))).fN;
+      LOG(debug) << "other.binContent(1, 1) = " << (*(dynamic_cast<IntensiveTH2F*>(other))).GetBinContent(1, 1);
+      LOG(debug) << "this->fN = " << fN;
+      LOG(debug) << "this->binContent(1, 1) = " << GetBinContent(1, 1);
+      for (int iX = 1; iX <= GetNbinsX(); iX++) {
+        for (int iY = 1; iY <= GetNbinsY(); iY++) {
+          SetBinContent(iX, iY, dynamic_cast<IntensiveTH2F*>(other)->GetBinContent(iX, iY));
+        }
+      }
```

Comment:
Perhaps you could also do `dynamic_cast` just once at the beginning and then make sure it is not `nullptr`?
Also, will you need these debug logs for the future?

Reply:
hi @knopers8 
I'm not sure how something can be `nullptr` in the merge method (asuming `this` and `other` are valid objects). But yes, you are right, it is better to use dynamic_cast once and check it for being `nullptr`.

Reply:
It is rather unlikely, but we've seen very unlikely events happening on grid, e.g. one of 10000 subjobs had memory issues, preventing ROOT to load correctly and generating corrupted objects. `other` could be invalid if it is `MergeInterface` but it is not `IntensiveTH2F`. It probably won't happen, but better safe than sorry.

Also, `dynamic_cast` has some overhead as compared to `reinterpret_cast`, so it's better to do it once for performance reasons.